### PR TITLE
test: TED OAuth Automation

### PR DESCRIPTION
**Pull request by Vijetha-Kaja**
## Description

- Automated TED OAuth 2.0 with authorization code

## How Has This Been Tested?
- Cypress test runs

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -189,6 +215,7 @@ jobs:
           docker run --name test-event-driver -d -p 22:22 -p 5001:5001 -p 3306:3306 \
           -p 5432:5432 -p 28017:27017 -p 25:25 -p 5000:5000 -p 3001:3000 --privileged --pid=host --ipc=host --volume /:/host -v ~/git-server/keys:/git-server/keys \
           -v ~/git-server/repos:/git-server/repos  appsmith/test-event-driver:latest
+          docker run --name appsmithted-oauth -d -p 6001:6001 appsmith/test-event-driver:nightly
```
Comment (Aishwarya-U-R) [inline .github/workflows/ci-test.yml:218]: Revert this file changes & try CI run now.

File: app/client/cypress/support/Pages/DataSources.ts [hunk 2]
```diff
@@ -900,4 +906,127 @@ export class DataSources {
...
+      clientId = clientId[1].split("<strong>client_secret: </strong>");
+      clientSecret = clientId[1].split("<strong>");
+      clientSecret = clientSecret[0].trim();
+      clientId = clientId[0].trim();
+      cy.wrap(clientId).as("OAuthClientID");
+      cy.wrap(clientSecret).as("OAuthClientSecret");
+    });
+  }
+
+  public CreateOAuthDatasource(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:951]: Call this method a part of CreateDataSource() - adding another parameter for OAuth.

File: app/client/cypress/support/Pages/DataSources.ts [hunk 3]
```diff
@@ -900,4 +906,127 @@ export class DataSources {
...
+    clientId: string,
+    clientSecret: string,
+  ) {
+    this.NavigateToDSCreateNew();
+    //Click on Authenticated API
+    cy.get(this._authApiDatasource).click({ force: true });
+    this.FillAPIOAuthForm(datasourceName, grantType, clientId, clientSecret);
+
+    // save datasource
+    cy.wait(500);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:963]: Use agHelper.Sleep(500);

File: app/client/cypress/support/Pages/DataSources.ts [hunk 4]
```diff
@@ -900,4 +906,127 @@ export class DataSources {
...
+    clientSecret: string,
+  ) {
+    if (dsName) {
+      // Change the Datasource name
+      cy.get(".t--edit-datasource-name").click();
+      cy.get(".t--edit-datasource-name input")
+        .clear()
+        .type(dsName, { force: true })
+        .should("have.value", dsName)
+        .blur();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:987]: Use  this.agHelper.RenameWithInPane(dataSourceName, false);

File: app/client/cypress/support/Pages/DataSources.ts [hunk 5]
```diff
@@ -900,4 +906,127 @@ export class DataSources {
...
+      datasourceFormData["OAUth_AccessTokenUrl"],
+    );
+    this.agHelper.TypeText(this._scope, "profile");
+    this.agHelper.TypeText(
+      this._authorizationURL,
+      datasourceFormData["OAuth_AuthUrl"],
+    );
+  }
+
+  public CreateAndFillApiAfterDSSaved(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:1014]: Why apiPage.CreateAndFillApi() not used?

Comment (Vijetha-Kaja) [reply app/client/cypress/support/Pages/DataSources.ts:1014]: Create and fill creates it directly from query pane but i need it created from the created data source

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/DataSources.ts:1014]: I see. Ok, so I see only line # 1021 click creates query from either newly created DS or from EntityExplorer. Lets add a parameter to CreateAndFillApi() & use the same method instead.

File: app/client/cypress/support/Pages/DataSources.ts (hunk 2 above)
Comment (Vijetha-Kaja) [reply app/client/cypress/support/Pages/DataSources.ts:950]: CreateOAuthDatasource() cannot be perfectly integrated to CreateDataSource() , especially saving the datasource , OAuth goes through a different flow for save

File: app/client/cypress/fixtures/datasources.json [hunk 6]
```diff
@@ -34,10 +34,19 @@
...
   "redshift-username": "root",
   "redshift-password": "Redshift$123",
+
   "smtp-host":"host.docker.internal",
   "smtp-port":"25",
   "smtp-username":"root",
   "smtp-password":"root",
+
+  "OAuth_username":"testuser@appsmith.com",
+  "OAuth_host":"http://localhost:6001",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/fixtures/datasources.json:44]: Use similar naming convention like other OAuth variables.

File: app/client/cypress/support/Pages/DataSources.ts [hunk 7]
```diff
@@ -900,4 +905,111 @@ export class DataSources {
...
+    );
+    this.agHelper.GetNClick(this._authType);
+    this.agHelper.GetNClick(this._oauth2);
+    this.agHelper.GetNClick(this._grantType);
+    if (grantType == "ClientCredentials")
+      this.agHelper.GetNClick(this._clientCredentails);
+    else if (grantType == "AuthCode")
+      this.agHelper.GetNClick(this._authorizationCode);
+
+    this.agHelper.UpdateInput(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/DataSources.ts:993]: Updated with UpdateInput(), to note.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**